### ISSUE TITLE
bug when surface isn't declared in schmidt calc

### ISSUE THREAD

**Issue by jread-usgs** [GLEON/rLakeAnalyzer#72]
```r
depth1 <- c(17.5, 20.5, 23.5, 24.5, 26.5, 29.5, 32.5, 40.5, 44.5, 48.5)
temp1 <- c(14.806071, 14.806071, 14.806071, 14.806071, 14.805789, 14.332454, 13.731522, 9.844114, 8.096372, 6.937981)

btha2=c(3750100,3504350,3297400,3106750,2914200,2706800,2511050,2315850,2133150,1958800,1791800,1634900,1486350,1327300,1170480,734940,419480,243160,10)
bthd2=c(0,1.5,3.5,5.5,7.5,9.5,11.5,13.5,15.5,17.5,19.5,21.5,23.5,25.5,27.5,33.5,38.5,43.5,48.5)

schmidt.stability(wtr=temp1,depths=depth1,bthA=btha2,bthD=bthd2)
Error in water.density(wtr, sal) : 
  water temperature array must be the same length as the salinity array
```

Looks like because when `sal` is missing it isn't evaluated until after wtr adds a value, and is therefore one too long:
```r
if (min(bthD) < depths[1]) {
    wtr = c(wtr[1], wtr)
    sal = c(sal[1], sal)
    depths = c(min(bthD), depths)
  }
```
So `sal` ends up longer than `wtr` in this situation. Can we just call `sal = c(sal[1], sal)` before the `wtr` line in that code block?

**Comment by jread-usgs** [comment]
This error no longer happens when version `1.8.4` is used. The CRAN version (and probably the version used by the person that reported this to me) is `1.8.3` which does produce this error.

**Comment by jread-usgs** [comment]
Fixed in https://github.com/GLEON/rLakeAnalyzer/commit/8e75ca95c715b41ea25ae7a07bcdcfebb581cdd7
